libs/newsletters-data-client/src/lib/newsletter-storage/InMemoryNewsletterStorage.ts

```diff
@@ -118,11 +118,10 @@ export class InMemoryNewsletterStorage implements NewsletterStorage {
 		}
 
 		this.memory.splice(this.memory.indexOf(match), 1);
-		const response: SuccessfulStorageResponse<NewsletterData> = {
+		return Promise.resolve({
 			ok: true,
-			data: match,
-		};
-		return Promise.resolve(response);
+			data: `newsletter #${listId} deleted`,
+		} as SuccessfulStorageResponse<string>);
 	}
```

Comment:
as below - I'd suggest returning an object rather than a string just to make the responses easier to handle.